libraries/Microsoft.Bot.Builder/IBotContext.cs

```diff
@@ -12,7 +12,7 @@ namespace Microsoft.Bot.Builder
 {
     public interface IBotContext
     {
-        Bot Bot { get; }
+        BotServer Bot { get; }
```

Comment:
Would developers get any benefit for providing their own implementation of a BotServer? If yes, maybe we could have the IBotContext depend on a IBotServer abstraction instead of a BotServer concrete (abstract) class.